### DISCUSSION TITLE
Any chance we could add an extra option to `swc.minify`

### DISCUSSION THREAD

**Discussion by IWANABETHATGUY** [swc-project/swc#6285] [Q&A]
https://github.com/swc-project/swc/blob/main/crates/swc/src/lib.rs#L924-L928

![image](https://user-images.githubusercontent.com/17974631/198562568-f5c86d26-c678-42fe-a770-caa7b2044fa3.png)
Any chance we could add an extra param `minfiy: bool`, sometimes we want to use `DCE` of swc `minify` without minifying the code,(for example development mode, DCE could improve the first screen time)
In such a way, we could pass the `minify` param to https://github.com/swc-project/swc/blob/main/crates/swc/src/lib.rs#L1068. I try to copy the `minify` and override it but failed because some part struct is private. (https://github.com/swc-project/swc/blob/main/crates/swc/src/lib.rs#L1019).

**Comment by IWANABETHATGUY** [comment]
Or would you mind if I open a pr to change the visibility of this field to `pub` https://github.com/swc-project/swc/blob/main/crates/swc/src/lib.rs#L1190

**Comment by kdy1** [comment]
Aren't you using it from rust? DCE is a separate pass, and I'm not sure why are you trying to use `swc` crate for this task

**Reply by IWANABETHATGUY** [reply]
I found that I could rewrite the `IdentCollector`, seems the problem is solved.

**Reply by IWANABETHATGUY** [reply]
> Aren't you using it from rust? DCE is a separate pass, and I'm not sure why are you trying to use `swc` crate for this task

Currently, our minify phase doesn't reuse the ast, which means we need to stringify the ast and use `minify` in swc crate, of course, this is inefficient.
